lib/data/external/PfsClient.js

```diff
@@ -72,9 +67,15 @@ class PfsClient {
 
     delete(objectGetInfo, reqUids, callback) {
         const log = createLogger(reqUids);
-        logHelper(log, 'error', 'Not implemented', errors.NotImplemented,
+        const key = objectGetInfo.key ? objectGetInfo.key : objectGetInfo;
```

Comment:
Can `objectGetInfo` be a string? In which case I think it has to be checked explicitly, e.g. with `typeof objectGetInfo === "string"` to avoid attempting to access a field in a string.

Reply:
Probably, since that's how the AzureClient's delete handles it. Just updated the code to check for that.